conf/validation-config/validation-types.conf

```diff
@@ -29,7 +29,7 @@ validation-types {
     errorMsg: "ers.upload.error.last-name"
   }
   nino: {
-    rule: "[A-Z]{2}[0-9]{6}[A-Z]{0,1}"
+    rule: "[A-Z]{2}[0-9]{6}[A-Z]{1}"
```

Comment:
I will need to double check this with Muhammed/Danny, currently this change was just added to prevent the last character being left blank, more complicated changes are probably out of scope for this ticket but I'll get it double checked now.

Reply:
I think something like `^(?!BG)(?!GB)(?!NK)(?!KN)(?!TN)(?!NT)(?!ZZ)(?:[A-CEGHJ-PR-TW-Z][A-CEGHJ-NPR-TW-Z])(?:\s*\d\s*){6}([A-D])$` could work though I need to get the all clear first

Reply:
I would say you need to use this hmrc lib to make your life easier https://github.com/hmrc/domain/blob/main/src/main/scala/uk/gov/hmrc/domain/Nino.scala#L40

Reply:
Yeah I agree, though I'm not sure if I'm misreading the regex but this seems to allow certain non allowed characters like BG, GB etc from [gov.uk](https://www.gov.uk/hmrc-internal-manuals/national-insurance-manual/nim39110)

Reply:
Ah yeah I see, there is also a supplied `List("BG", "GB", "NK", "KN", "TN", "NT", "ZZ")` provided below. Though for the rule in this config file I'm not sure if there's any way of using the list, so it would likely have to be included in the regex like this `^(?!BG|GB|KN|NK|NT|TN|ZZ)[^DFIQUV][^DFIQUVO]\s?[0-9]{6}\s?[ABCD]$` @frikit

Reply:
@DannyPennington97 confirmed this will likely have to be left to a future ticket